documentation/decisions/adr/2023-12-18-measurement-dashboard-architecture.md

```diff
@@ -36,40 +36,95 @@ The goal of this ADR is to evaluate and recommend an architectural approach for
 
 ## Options Considered
 
-- Analytics API + user interface
+- S3 bucket + dashboard user interface
+- Analytics API + dashboard user interface
 - Custom dashboard application (e.g. [Dash][plotly-dash] or [R Shiny][r-shiny])
 - Open source dashboard solution (e.g. [Metabase][metabase] or [Redash][redash])
-- SaaS dashboard solution (e.g. [PowerBI][power-bi] or [Tableau][tableau])
+- SaaS dashboard solution (e.g. [PowerBI][power-bi], [Tableau][tableau], [Looker][looker], [Amazon Quicksight][quicksight], or [Domo][domo])
+
+> [!NOTE]
+> This ADR does not aim to recommend a *specific* platform or tool, but rather a general architectural approach to building and publishing dashboards. Once we select an approach from the list above (e.g. open source dashboard solution), we'll need to create a separate ADR that evaluates the options available within that approach (e.g. Metabase vs Redash).
 
 ## Decision Outcome <!-- REQUIRED -->
 
-TODO: Decide between analytics API + UI and open source dashboard solution.
+### Long-term
+
+Our recommendation for a long-term dashboard architecture is to combine **analytics API + dashboard UI** with an **open source dashboard solution** in a way that allows us to quickly iterate on and test the value of new metrics or charts, then "promote" the ones we find most helpful to an official public dashboard.
+
+The steps for this promotion strategy would likely include:
+
+1. **ETL:** Write the data needed to calculate operational and program metrics to a data warehouse via an ETL pipeline.
+2. **Ad hoc report:** Prototype a new metric by building an ad hoc SQL report in an open source dashboard solution (e.g. Metabase or Redash) connected to that data warehouse.
+3. **Temporary dashboard:** If that ad hoc report is useful to stakeholders, then incorporate it into a temporary dashboard within the open source dashboard tool.
+4. **Public dashboard and endpoint:** Once we get feedback on this temporary dashboard, formalize the metric in a new (or modified) API endpoint that is consumed by a new (or modified) page in the public-facing dashboard application.
+
+### Short-term
+
+Because setting up the infrastructure for the long-term dashboard architecture will take a significant amount of time and resources, in the short-term, we recommend following the **s3 bucket + dashboard UI** option. This approach reduces the upfront infrastructure investment needed to publish an initial dashboard, while still laying the foundation for parts of the long-term approach.
+
+The steps for publishing a dashboard using this short-term strategy would likely include:
+
+1. **ETL:** Use a simple ETL pipeline (e.g. scheduled GitHub action) to:
+   1. Extract data needed to calculate operational and program metrics from external sources.
+   2. Calculate the metrics we want to add to a dashboard.
+   3. Load both the source data and the metrics results to s3 buckets.
```

Comment:
At some point in here, we should discuss making sure that any sensitive data (that either has PII or has privacy concerns due to small sample size, or that is considered unfit for public consumption) does not get put in S3. But we can put the aggregate statistics in S3 and build on top of them, as well as the raw data so long as the raw data is not inappropriate.

Reply:
Good call, I added a note here in this commit: https://github.com/HHS/simpler-grants-gov/pull/936/commits/39428b92fded127603ec7ea292086908260b6a51